libopae/src/event.c

```diff
@@ -680,14 +683,17 @@ fpga_result __FPGA_API__ fpgaUnregisterEvent(fpga_handle handle,
 	struct _fpga_event_handle *_eh = (struct _fpga_event_handle *) event_handle;
 	struct _fpga_token *_token;
 
-	result = handle_check_and_lock(_handle);
-	if (result)
-		return result;
-
+	// lock event handle first as other event API functions only lock the
```

Comment:
Same comment as above re: the ordering.  Not sure why this order would avoid a deadlock.